src/mesh/index_derivs.cxx

```diff
@@ -1251,7 +1269,13 @@ const Field3D Mesh::indexDDY(const Field3D &f, CELL_LOC outloc, DIFF_METHOD meth
 
   result.setLocation(diffloc); // Set the result location
 
-  return interp_to(result, outloc); // Interpolate if necessary
+  // return interp_to(result, outloc); // Interpolate if necessary
+  // Cannot interpolate result without communicating and applying boundary
+  // conditions first, instead check that it is at outloc
+  ASSERT2(((outloc == CELL_DEFAULT) && (result.getLocation() == f.getLocation())) ||
+          (result.getLocation() == outloc));
```

Comment:
Can these checks be moved earlier? i.e. what's the earliest we can bail out if something's wrong?

Reply:
Looking at them again, I think these checks only replicate the inloc/outloc checks already done at the top of the functions, so I have just deleted them.